tests/marray/marray_common.h

```diff
@@ -0,0 +1,113 @@
+//
+//  Copyright (c) 2022 The Khronos Group Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+*******************************************************************************/
+
+#ifndef SYCL_CTS_TEST_MARRAY_MARRAY_COMMON_H
+#define SYCL_CTS_TEST_MARRAY_MARRAY_COMMON_H
+
+#include "../common/common.h"
+#include "../common/type_coverage.h"
+#include "../common/type_list.h"
+#include "marray_custom_type.h"
+
+namespace marray_common {
+
+/** @brief Execute the tests with the same values for NumElements. */
+inline auto get_num_elements() {
+  return value_pack<int, 1, 2, 3, 4, 5, 8, 16, 32, 64>::generate_unnamed();
+}
+
+/** @brief Execute the tests with the same values for DataT. */
+inline auto get_types() {
+#ifdef SYCL_CTS_COMPILING_WITH_COMPUTECPP
+  WARN(
+      "ComputeCPP does not support custom types other than sycl::half."
+      "Skipping the test case for custom types.");
+#endif
+  return named_type_pack<
+      char, int, float, typename std::int8_t, typename std::int32_t
+  // does not work with any custom type other than sycl::half
+#ifndef SYCL_CTS_COMPILING_WITH_COMPUTECPP
+      ,
+      custom_int
+#endif
+#if SYCL_CTS_ENABLE_FULL_CONFORMANCE
+      ,
+      unsigned char, short, unsigned short, unsigned int, long, unsigned long,
+      long long, unsigned long long, bool, typename std::uint8_t,
+      typename std::int16_t, typename std::uint16_t, typename std::uint32_t,
+      typename std::int64_t, typename std::uint64_t
+#endif  // SYCL_CTS_ENABLE_FULL_CONFORMANCE
+      >::generate("char", "int", "float", "std::int8_t", "std::int32_t"
+#ifndef SYCL_CTS_COMPILING_WITH_COMPUTECPP
+                  ,
+                  "custom_type"
+#endif
+#if SYCL_CTS_ENABLE_FULL_CONFORMANCE
+                  ,
+                  "unsigned char", "short", "unsigned short", "unsigned int",
+                  "long", "unsigned long", "long long", "unsigned long long",
+                  "bool", "std::uint8_t", "std::int16_t", "std::uint16_t",
+                  "std::uint32_t", "std::int64_t", "std::uint64_t"
+#endif  // SYCL_CTS_ENABLE_FULL_CONFORMANCE
+  );
+}
+
+template <typename DataT, std::size_t NumElements>
+struct helper {
+  using marray_t = sycl::marray<DataT, NumElements>;
+
+  /** @brief Initialize using the sequence 1, 2, 3, etc. */
+  static sycl::marray<DataT, NumElements> init_inc() {
+    sycl::marray<DataT, NumElements> ret;
+    for (std::size_t i = 0; i < NumElements; i++) {
```

Comment:
This looks like iota algorithm.

Reply:
I have replaced the `init_inc` function by `std::iota` to make the code simpler.